WcaOnRails/db/schema.rb

```diff
@@ -1120,6 +1145,8 @@
 
   add_foreign_key "active_storage_attachments", "active_storage_blobs", column: "blob_id"
   add_foreign_key "active_storage_variant_records", "active_storage_blobs", column: "blob_id"
+  add_foreign_key "groups", "groups", column: "parent_group_id"
+  add_foreign_key "roles", "groups"
```

Comment:
It's a bit curious that the foreign key for roles -> users is not listed here. Can you run `bin/rails db:schema:dump:primary` again to make sure that this is really intended by Rails? (and also make sure that the foreign key is actually present in your local DB and you're not on some work-in-progress state where the FK isn't there yet)?

Reply:
I tried running, but it didn't come. When I checked my local sql, even there the foreign key is not mentioned. Can it be because I did something wrong in migration file?

Reply:
Maybe it's because in your migration file, you're listing `t.references` for most foreign keys but `t.integer :user_id` for users. Also see my other comment directly in the migration file

Reply:
Yeah, sometimes there are legacy problems like the one you described. My guess is that Rails used to use `integer` as a default value at some point in time, and then switched to `bigint` later. Now tables that have been created long ago still use `integer` ID columns, and the reference type has to be changed accordingly.